portfolio/portfolio.go

```diff
@@ -27,55 +28,74 @@ const (
 	ExchangeAddress = "Exchange"
 	// PersonalAddress is a label for a personal/offline address
 	PersonalAddress = "Personal"
+
+	defaultAPIKey = "Key"
+)
+
+var (
+	errNotETHAddress           = errors.New("not an Ethereum address")
+	errAddressIsEmptyOrInvalid = errors.New("address is empty or invalid")
+	errProviderNotFound        = errors.New("provider not found")
+	errProviderNotEnabled      = errors.New("provider not enabled")
+	errProviderAPIKeyNotSet    = errors.New("provider API key not set")
+	errPortfolioItemNotFound   = errors.New("portfolio item not found")
+	errUnsupportedCoinType     = errors.New("unsupported coin type")
+	errNoPortfolioItemsToWatch = errors.New("no portfolio items to watch")
 )
 
-var errNotEthAddress = errors.New("not an Ethereum address")
+// GetEthereumAddressBalance fetches Ethereum address balance for a given address
+func (b *Base) GetEthereumAddressBalance(ctx context.Context, address string) (float64, error) {
+	if ok, _ := common.IsValidCryptoAddress(address, "eth"); !ok {
```

Comment:
Bringing up unrelated things sorry. What makes this address invalid?
https://etherscan.io/address/0xdadb0d80178819f2319190d340ce9a924f783711

```
[ERROR] | PORTFOLIO | 26/03/2025 10:59:59 | PortfolioWatcher: UpdatePortfolio error: error getting balance for 0xdadB0d80178819F2319190D340ce9A924f783711: not an Ethereum address for currency ETH
```